packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -143,28 +360,91 @@ class HttpGatewayRouteSpec extends GatewayRouteSpec {
    */
   readonly routeType: Protocol;
 
+  /**
+   * The default target host name to write to.
+   * Boolean is used to select either true = enable or  false = disable.
+   * When enabled, rewrites the original request received at the Virtual Gateway to the destination Virtual Service name.
+   * When disabled, retains the original hostname/prefix from the request.
+   *
+   * @default true
+   */
+  readonly rewriteRequestHostname?: boolean;
+
+  /**
+   * The path to rewrite.
+   *
+   * @default - rewritten to the target Virtual Service's hostname and the matched prefix is rewritten to '/'.
+   */
+  readonly pathRewrite?: HttpGatewayRoutePathRewrite;
+
   constructor(options: HttpGatewayRouteSpecOptions, protocol: Protocol.HTTP | Protocol.HTTP2) {
     super();
     this.routeTarget = options.routeTarget;
     this.routeType = protocol;
     this.match = options.match;
+    this.rewriteRequestHostname = options.rewriteRequestHostname;
+    this.pathRewrite = options.pathRewrite;
   }
 
-  public bind(_scope: Construct): GatewayRouteSpecConfig {
-    const prefixPath = this.match ? this.match.prefixPath : '/';
-    if (prefixPath[0] != '/') {
-      throw new Error(`Prefix Path must start with \'/\', got: ${prefixPath}`);
+  public bind(scope: Construct): GatewayRouteSpecConfig {
+    const pathRewriteConfig = this.pathRewrite?.bind(scope);
+    const pathMatchConfig = this.match?.path?.bind(scope).requestMatch;
+
+    // Set prefix Match to '/' if none on match properties are defined.
+    const prefixMatch = areMatchPropertiesUndefined(this.match)
+      ? '/'
+      : pathMatchConfig?.prefix;
+    const prefixPathRewrite = pathRewriteConfig?.prefix;
+    const pathRewrite = pathRewriteConfig?.path;
+
+    if (prefixMatch && prefixMatch[0] != '/') {
+      throw new Error(`Prefix Path for the match must start with \'/\', got: ${prefixMatch}`);
+    }
+    if (prefixPathRewrite?.value && prefixPathRewrite.value[0] != '/') {
+      throw new Error(`Prefix Path for the rewrite must start with \'/\', got: ${prefixPathRewrite.value}`);
     }
+    if (prefixPathRewrite && !prefixMatch) {
+      throw new Error('HTTP Gateway Route Prefix Match must be set.');
```

Comment:
I don't understand what this error means... maybe something like

```suggestion
      throw new Error('HTTP Gateway Route prefix match must be provided if a rewrite prefix was provided');
```

?

Reply:
Thanks. Changed as suggested.

Reply:
Actually changed to "HTTP Gateway Route prefix match must be provided if a **prefix rewrite** was provided"